src/main/kotlin/sliv/tool/catalogue/view/CatalogueView.kt

```diff
@@ -1,10 +1,171 @@
+            fileNameViewRadioButton = radiobutton("File view", viewFormatToggleGroup)
+            imagePreviewRadioButton = radiobutton("Image preview", viewFormatToggleGroup)
+        }
+        bottom {
+            hbox(alignment = Pos.CENTER) {
+                button("Apply") {
+                    action {
+                        controller.createFramesSelection(fileNamesListView.selectedItems().map { it.frame })
+                    }
+                }
+            }
+        }
+    }
+
+    override val root = catalogueBorderpane.also { initializeNodes() }
+
+    private fun onViewFormatToggleSwitched(switchedToToggle: Toggle) {
+        val selectedViewFormat = when (switchedToToggle as RadioButton) {
+            fileNameViewRadioButton -> ViewFormat.FILE_NAME
+            imagePreviewRadioButton -> ViewFormat.IMAGE_PREVIEW
+            else -> ViewFormat.FILE_NAME.also { println("Unexpected view format toggle!") }
+        }
+        updateViewFormatNode(selectedViewFormat)
+    }
+
+    private fun getViewFormatNode(withFormat: ViewFormat) = when (withFormat) {
+        ViewFormat.FILE_NAME -> fileNamesListView
+        ViewFormat.IMAGE_PREVIEW -> null // TODO("Add an image preview format")
+    }
+    private fun updateViewFormatNode(withFormat: ViewFormat) {
+        catalogueBorderpane.center = getViewFormatNode(withFormat)
+    }
+
+    private fun initializeNodes() {
+        initializeViewFormatToggles()
+        initializeSelectionNodes()
+    }
+
+    private fun getViewFormatToggle(viewFormat: ViewFormat) = when (viewFormat) {
+        ViewFormat.FILE_NAME -> fileNameViewRadioButton
+        ViewFormat.IMAGE_PREVIEW -> imagePreviewRadioButton
+    }
+
+    private fun initializeViewFormatToggles() {
+        viewFormatToggleGroup.selectedToggleProperty().onChange {
+            it ?: return@onChange
+            onViewFormatToggleSwitched(it)
+        }
+        viewFormatToggleGroup.selectToggle(getViewFormatToggle(currentViewFormat))
+    }
+
+    private fun setSelectionCheckBoxState(selectionState: SelectionState) {
+        when (selectionState) {
+            SelectionState.ALL -> {
+                selectionCheckBox.isIndeterminate = false
+                selectionCheckBoxBoolProperty.value = true
+            }
+            SelectionState.NONE -> {
+                selectionCheckBox.isIndeterminate = false
+                selectionCheckBoxBoolProperty.value = false
+            }
+            SelectionState.PART -> selectionCheckBox.isIndeterminate = true
+        }
+    }
+
+    private fun haveFieldDoubleClick() = previousSelectedField != null && previousSelectedField == currentSelectedField
+
+    private fun initializeSelectionNodes() {
+        fileNamesListView.setOnMouseClicked {
+            if (haveFieldDoubleClick() && previousSelectedFieldsCount == 1) {
+                fileNamesListView.deselectAllItems()
+                fileNamesListView.selectItem(currentSelectedField)
```

Comment:
what is done here?

Reply:
I add an unselect logic for a single selected field on double click. It will be easier to show visually.
before: 
![catalogue_double_click_before](https://user-images.githubusercontent.com/66365072/203765812-fbce77d5-ac82-4705-8926-d0aa1d5b9641.gif)
after:
![catalogue_double_click_after](https://user-images.githubusercontent.com/66365072/203765856-eba166c3-50b7-4865-a4cf-45f5ec66302e.gif)

Reply:
why is it required? looks like IDEA or explorer cant do this